yax/shared/utilities/taxidtool.py

```diff
@@ -0,0 +1,445 @@
+import sys
+
+
+def preprocessing(nodes_fp, names_fp, gi_taxid_nucl_fp, output_location,
+                  project_name):
+    ##########################################################################
+    # This module, accessed with the tag -build, is responsible for compiling
+    # all pertinent information into a single file that can be quickly read by
+    # later Tax Tree Clipping actions.
+    #
+    # Input
+    #       nodes_fp    STRING  location of nodes.dmp file
+    #       names_fp    STRING  location of names.dmp file
+    #       gi_taxid_nucl_fp    STRING  location of gi_taxid_nucl.dmp file
+    #       output_location     STRING  desired output location of taxid_data
+    #                                   file
+    #
+    # Output
+    #       taxid_data  FILE    file containing a line for each taxid and it's
+    #                           associated information
+    ###########################################################################
```

Comment:
This is good, but let's follow [numpydoc](https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt) convention, which has tooling support.